src/resolvers/create_session.js

```diff
@@ -71,12 +72,33 @@ export default async (
     `
   } else {
     // session was not found so we create a new one
+
+    const parsed_payload = () => {
+      try {
+        return JSON.parse(payload)
+      } catch {
+        return {}
+      }
+    }
+
+    MAIL.send([
+      MAIL.NEW_SESSION,
+      user.uuid,
+      mail,
+      JSON.stringify({
+        ...parsed_payload(),
+        browser: session.browserName,
+        os_name: session.osName,
+        device : session.deviceModel,
+      }),
+    ])
```

Comment:
The auth should never read the payload, you can send session fields in others arguments and simply forward the payload